File: src/main/java/ru/mm/surv/codecs/webm/incubator/streamm/MovieFragment.java
package ru.mm.surv.codecs.webm.incubator.streamm;

import ru.mm.surv.codecs.webm.util.stream.Buffer;

/**
 *
 * @author Varga Bence (vbence@czentral.org)
 */
public interface MovieFragment {

    int LIMIT_FRAME_MINIMUM = 100 * 1024;

    int LIMIT_FRAME_MAXIMUM = 2048 * 1024 + 64 * 1024;

    Buffer[] getBuffers();

    int length();

}

File: src/main/kotlin/ru/mm/surv/codecs/webm/incubator/streamm/MovieFragment.kt
package ru.mm.surv.codecs.webm.incubator.streamm

import ru.mm.surv.codecs.webm.util.stream.Buffer

/**
 *
 * @author Varga Bence (vbence@czentral.org)
 */
interface MovieFragment {
    fun getBuffers(): Array<Buffer>
    fun length(): Int

    companion object {
        const val LIMIT_FRAME_MINIMUM = 100 * 1024
        const val LIMIT_FRAME_MAXIMUM = 2048 * 1024 + 64 * 1024
    }
}